handbook.md

```diff
@@ -0,0 +1,233 @@
+* A plugin, by taking care of documentation, code reviews, and packages to foreman-packaging
+* Debian packaging, by looking at which projects need debian packages and updating them
+* A subsystem, such as a Compute Resource or an Operating System in Foreman and updating it properly.
+
+Traditionally, the process to become a maintainer has been simple. Take care of something for long enough and people will expect you to maintain it. We might ask you directly if you want to be a maintainer if you're maintaining something for some time. If you have made contributions only on a very specific topic (an Operating System, for instance), we might ask you to be a maintainer for that part of the code.
+
+## API usability and versioning
+
+We follow [Semantic versioning 2.0](http://semver.org/) for anything above '0.x' releases. This means incompatible changes, such as breaking an API endpoint, not working with a previous Foreman version (in the case of a plugin), require increasing the major version. The bulk of your changes should only require increasing the minor and patch versions, for backwards-compatible functionality and bugfixes.
+
+We strive to remain compatible and not break other people's workflows, so if you have made some breaking changes that will need increasing the major version, please use `Foreman::Deprecation` to deprecate anything and warn in which version will it be removed. It supports two calls:
+
+`Foreman::Deprecation.api_deprecation_warning(message)`
+
+`Foreman::Deprecation.deprecation_warning(foreman_version_deadline, message)`
+
+## Architectural decisions
+
+### What exactly is core?
+
+* Features that most of our userbase use. We get this information through surveys.
+* Abstractions such as the Fact importer and parser, Power manager, etcetera.
+* API
+* A framework for plugin development.
+* Compute Resources, Host groups, Reports, UI helpers, Nics, Host, Smart proxies are a few examples of key models in core.
+
+### What do we pull from core to make it a plugin?
+
+* Features we do not want to maintain, either because they are complex and not widely used, or because they are difficult to test and maintain.
+* Very specific features that most of our user base does not use.
+
+Examples of things we want to take out are Puppet, Host group based provisioning, GCE compute resource, APIv1.
+
+### Why some features might start as plugins?
+
+The controls in core, such as deep code reviews, plus being subject to quarterly releases, makes it difficult to quickly develop some features. For this reason, you might decide to start off your feature as a plugin. Then, once your code has been released, tested, and after review, it could be pulled into core if deemed appropriate. In any case, if your feature is very widely used, it could simply become a default on new installations if it doesn't fit in well in core.
```

Comment:
I don't think there's much precedent for suggesting features should become a default on new installations.